src/main/java/jalf/relation/algebra/Intersect.java

```diff
@@ -67,4 +70,10 @@ protected RelationType typeCheck() {
         return visitor.visit(this);
     }
 
+
+    @Override
+    protected Keys lazyComputeKey() {
+        return left.getKeys().union(right.getKeys());
```

Comment:
ok, provided we use `union({k1,k2},{k3,k4}) = {k1,k2,k3,k4}`, which is not the case right now